### PR TITLE
fix(metric): service.push does not throw error

### PR DESCRIPTION
**Pull request by ZhuzhginNA**
BREAKING CHANGE: MetricService requires ILogger

### CODE REVIEW

**Review by Demian222 — APPROVED** [review]

File: packages/metric/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "1.6.1",
   "description": "Metric module for nestjs enterprise",
   "type": "module",
+  "types": "./target/esm/index.d.ts",
```
Comment (oljekechoro) [inline packages/metric/package.json:6]: types should be only in publishConfig

File: packages/metric/package.json [hunk 2]
```diff
@@ -23,6 +24,11 @@
     ]
   },
   "exports": "./src/main/ts/index.ts",
+  "files": [
```
Comment (oljekechoro) [inline packages/metric/package.json:27]: files should be only in publishConfig

**Review by oljekechoro — CHANGES_REQUESTED** [review]

File: packages/metric/qiwi-nestjs-enterprise-metric-1.6.1.tgz
Comment (antongolub) [inline packages/metric/qiwi-nestjs-enterprise-metric-1.6.1.tgz:1]: ?

File: packages/metric/src/main/ts/metric.module.ts [hunk 3]
```diff
@@ -18,12 +18,16 @@ import { IGraphiteService } from './graphite.servise.interface'
...
+        logger: ILogger,
+      ) => {
         const options = {
           prefix: config.get('metric.prefix'),
           interval: config.get('metric.interval'),
         }
-        return new MetricService(graphiteService, options)
+        return new MetricService(graphiteService, options, logger)
       },
       inject: ['IGraphiteService', 'IConfigService'],
```
[the remark this answers is not in the document]
Comment (oljekechoro) [reply packages/metric/src/main/ts/metric.module.ts:32]: Do not forget

File: packages/metric/src/main/ts/metric.module.ts [hunk 4]
```diff
@@ -49,8 +53,8 @@ export class MetricModule {
         },
         {
           provide: 'IMetricService',
-          useFactory: (graphiteService) => {
-            return new MetricService(graphiteService, metricsConfig)
+          useFactory: (graphiteService, logger) => {
+            return new MetricService(graphiteService, metricsConfig, logger)
           },
           inject: ['IGraphiteService'],
```
Comment (oljekechoro) [inline packages/metric/src/main/ts/metric.module.ts:59]: ILogger is missing too

**Review by oljekechoro — CHANGES_REQUESTED** [review]

**Review by oljekechoro — APPROVED** [review]

**Pull request merged**